From 71f912ffc57548c8f47f1567844ed41a5f05bc83 Mon Sep 17 00:00:00 2001
From: Bart Ribbers <bribbers@disroot.org>
Date: Wed, 16 Sep 2020 10:44:06 +0200
Subject: [PATCH] Fix compilation on Musl

Without this fix, compilation failed on Musl due to ssize_t being an
unknown type

In file included from ../src/xrt/state_trackers/prober/p_prober.c:15:
../src/xrt/state_trackers/prober/p_prober.h:67:2: error: unknown type name 'ssize_t'
   67 |  ssize_t interface;
      |  ^~~~~~~
---
 src/xrt/state_trackers/prober/p_prober.h | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/xrt/state_trackers/prober/p_prober.h b/src/xrt/state_trackers/prober/p_prober.h
index bf7a3417..793c14a4 100644
--- a/src/xrt/state_trackers/prober/p_prober.h
+++ b/src/xrt/state_trackers/prober/p_prober.h
@@ -23,6 +23,9 @@
 #include <libuvc/libuvc.h>
 #endif
 
+#ifndef __KERNEL__
+#include <sys/types.h>
+#endif
 
 /*
  *
-- 
GitLab

